sw/applications/example_asm/main.c

```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "constants.h"
+#include "x-heep.h"
+
+/* By default, printfs are activated for FPGA and disabled for simulation. */
+#define PRINTF_IN_FPGA  1
+#define PRINTF_IN_SIM   0
+
+#if TARGET_SIM && PRINTF_IN_SIM
```

Comment:
`#if TARGET_SIM && PRINTF_IN_SIM
        #define PRINTF(fmt, ...)    printf(fmt, ## __VA_ARGS__)
#elif PRINTF_IN_FPGA
    #define PRINTF(fmt, ...)    printf(fmt, ## __VA_ARGS__)
#else
    #define PRINTF(...)
#endif`

Reply:
yes, we are porting more and more boards, so actually adding just the pynq is not OK, you should be adding also the zcu and the nexys. To avoid that (for now...) let's leave it as general as the sniped i pasted